detection_rules/rule.py

```diff
@@ -583,6 +583,7 @@ class QueryRuleData(BaseRuleData):
     index: Optional[List[str]]
     query: str
     language: definitions.FilterLanguages
+    filters: Optional[List[dict]]
```

Comment:
Based on your examples, I also think we need to flush out the schema more with a dataclass.